dbms/src/Storages/KVStore/MultiRaft/ApplySnapshot.cpp

```diff
@@ -115,6 +117,21 @@ void KVStore::checkAndApplyPreHandledSnapshot(const RegionPtrWrap & new_region,
     }
 
     onSnapshot(new_region, old_region, old_applied_index, tmt);
+
+    if (tmt.getContext().getSharedContextDisagg()->isDisaggregatedStorageMode())
+    {
+        auto fap_ctx = tmt.getContext().getSharedContextDisagg()->fap_context;
+        // Everytime we meet a legacy snapshot, we try to clean obsolete fap ingest info.
+        if constexpr (!std::is_same_v<RegionPtrWrap, RegionPtrWithCheckpointInfo>)
+        {
+            fap_ctx->forceCleanCheckpointIngestInfo(tmt, new_region->id());
```

Comment:
Clean obsolete phase 1 result, when received another legacy snapshot.
Better firstly cancel fap process, however, there is no case currently where a legacy snapshot runs with fap phase1/phase2 in parallel,